vscode-trace-extension/src/trace-explorer/trace-tree.ts

```diff
@@ -146,15 +159,46 @@ export const fileHandler = (analysisTree: AnalysisProvider) => async (context: v
             }
         }
 
+        if (tracesArray.length === 0) {
+            progress.report({ message: 'Complete', increment: 100 });
+            traceLogger.addLogMessage('No valid traces found in the selected directory: ' + resolvedTraceURI, fileHandler.name);
+            panel.dispose();
+            return;
+        }
+
+        progress.report({ message: 'Opening trace(s)', increment: 20 });
         const traces = new Array<TspTrace>();
+        const invalidTraces = new Array<string>();
         for (let i = 0; i < tracesArray.length; i++) {
             const traceName = tracesArray[i].substring(tracesArray[i].lastIndexOf('/') + 1);
             const trace = await traceManager.openTrace(tracesArray[i], traceName);
             if (trace) {
                 traces.push(trace);
+            } else {
+                traceLogger.addLogMessage('Failed to open trace: ' + traceName, fileHandler.name);
+                traceLogger.addLogMessage('There may be an issue with the server or the trace is invalid.', fileHandler.name);
+                invalidTraces.push(traceName);
             }
         }
 
+        if (token.isCancellationRequested) {
+            // Rollback traces
```

Comment:
I'd remove this one-liner comment, as we already state the matter thanks to line 186's message right below.